src/js/components/projects/createForm.tsx

```diff
@@ -74,10 +89,134 @@ const ProjectForm = ({
     resetForm();
   };
 
+  const handleChange = (checked: boolean) => {
+    if (checked) {
+      setFromBranchChecked(true);
+    } else {
+      setFromBranchChecked(!fromBranchChecked);
+    }
+  };
+
+  const doGetBranches = async () => {
+    if (repoBranches.length) {
+      setBranchMenuOpen(true);
+    } else {
+      if (branchMenuOpen) {
+        setBranchMenuOpen(false);
+      }
+      setFetchingBranches(true);
+      // fetching feature branches here when option selected,
+      // in lieu of storing in store...
+      const baseBranches = await apiFetch({
+        url: `${window.api_urls.repository_detail(
+          repository.id,
+        )}/feature_branches`,
+        dispatch,
+      });
+      setRepoBranches(baseBranches);
+      setFetchingBranches(false);
+      setBranchMenuOpen(true);
+    }
+  };
+
+  const handleBranchSelection = (selection: any) => {
+    setInputs({ ...inputs, ...inputs.branch_name, branch_name: '' });
+    setBaseBranch(selection[0].label);
+    setBranchMenuOpen(false);
+  };
+
+  const resetBranch = (e: React.MouseEvent<HTMLButtonElement>) => {
+    e.preventDefault();
+    setInputValue('');
+    setBranchMenuOpen(false);
+    setFromBranchChecked(false);
+  };
+
+  // eslint-disable-next-line @typescript-eslint/camelcase
+  const handleBranchChange = (value: string) => {
+    if (baseBranch) {
+      setBaseBranch(inputs.branch_name);
+    }
+    setInputs({ ...inputs, ...inputs.branch_name, branch_name: value });
+  };
+  const noOptionsFoundText = (
+    <p data-form="project-create">
+      {i18n.t("There aren't any available branches at this time.")}{' '}
+      <Button
+        label={i18n.t('Check Again')}
+        variant="base"
+        onClick={doGetBranches}
+      />
+      {i18n.t('to refresh this list or')}{' '}
+      <Button
+        label={i18n.t('start from a new branch')}
+        variant="link"
+        onClick={resetBranch}
+      />
+      .
+    </p>
+  );
+
+  const branchOptions = repoBranches.map((item, index) => ({
+    id: `${index + 1}`,
+    label: item,
+  }));
+
   return (
     <form onSubmit={handleSubmit} className="slds-form slds-m-bottom--large">
       {isOpen && (
         <>
+          <Checkbox
+            id="project-base-branch"
+            assistiveText={{
+              label: `${i18n.t('Use existing Github branch')}`,
+            }}
+            labels={{
+              label: `${i18n.t('Use existing Github branch')}`,
+            }}
+            checked={fromBranchChecked}
+            className="slds-form-element_stacked slds-p-left_none"
+            onChange={(
+              event: React.ChangeEvent<HTMLInputElement>,
+              { checked }: { checked: boolean },
+            ) => handleChange(checked)}
+          />
+          {fromBranchChecked && (
+            <Combobox
+              id="combobox-inline-single"
+              isOpen={branchMenuOpen}
```

Comment:
I'm really surprised we'd need to handle the open/closed status of the menu manually. I don't see that in any of the examples on https://react.lightningdesignsystem.com/components/comboboxes/ and the menus seem to open and close naturally.

Reply:
@jgerigmeyer I think this was added to prevent opening on focus;  since the options don't get populated until then, it would display the empty options text first while fetching the feature branches.

Reply:
I think that's probably okay though, especially if we use `hasInputSpinner` and/or `hasMenuSpinner`? Also if we fetch on checkbox-click rather than input-focus.

Reply:
@jgerigmeyer since we made the change to fetch feature branches when the checkbox is checked, I found it necessary to keep manually open/close the menu; otherwise it didn't seem to open once the request completed.